util/sort.lua

```diff
@@ -72,34 +75,54 @@ end
 ---@param b Section
 ---@return boolean
 function sort.SortSectionsBySizeDescending(a, b)
-  return a:GetCellCount() > b:GetCellCount()
+  local aSize, bSize = a:GetCellCount(), b:GetCellCount()
+  if aSize ~= bSize then
+    return aSize > bSize
+  end
+  return a.title:GetText() < b.title:GetText()
 end
 
 ---@param a Section
 ---@param b Section
 ---@return boolean
 function sort.SortSectionsBySizeAscending(a, b)
-  return a:GetCellCount() < b:GetCellCount()
+  local aSize, bSize = a:GetCellCount(), b:GetCellCount()
+  if aSize ~= bSize then
+    return aSize < bSize
+  end
+  return a.title:GetText() < b.title:GetText()
 end
 
 ---@param a Item
 ---@param b Item
 ---@return boolean
 function sort.SortItemsByQualityThenAlpha(a, b)
   if invalidData(a.data, b.data) then return false end
-  if a.data.itemInfo.itemQuality == b.data.itemInfo.itemQuality then
+  if a.data.itemInfo.itemQuality ~= b.data.itemInfo.itemQuality then
+    return a.data.itemInfo.itemQuality > b.data.itemInfo.itemQuality
+  elseif a.data.itemInfo.currentItemLevel ~= b.data.itemInfo.currentItemLevel then
```

Comment:
Remove item level from the sort here, as it fundamentally changes the definition of this sort, as it's sorting on a new property that is not described in the name of the sort. Adding itemCount is okay though, because it's not a property of the item, but rather a property of the set.

If you'd like, add a new sorting function and all required options to support a new sort, named appropriately.

Reply:
Perhaps just swapping name and currentItemLevel here? I think item level is still useful as a tiebreaker even in this sort - as it would then only be used if the quality and item name were the same.

For example after spamming some dungeons I now have 3 of these:
![image](https://github.com/Cidan/BetterBags/assets/20693723/05bcf239-b2ea-48c5-98d1-6bd2479b6f46)

Without sorting by item level it's just falling back to GUID (since gear is always stack 1), which is less related to quality than item level. Let me know what you think.

Reply:
The user must always be aware of what the sort is doing and it must be communicated clearly. The issue here isn't a flaw in the fundamental approach, but the way we present the approach to the user. I don't necessarily disagree with you, but the user must always know exactly how their items are sorted.

I'm not opposed to renaming the functions, and their related configuration entries, to clearly communicate what is happening in the sort.

Reply:
I'll remove item level from the sort for now instead of increasing the scope of this PR beyond my original intention.